### PR TITLE
Activate window on unhide

### PR DESCRIPTION
**Pull request by wmww**
I'm not quite sure why I originally wrote this, but I just found it lying around in an old branch and it seemed like good code. The idea is a window shouldn't be able to steal focus from another app by unhiding an existing surface, but it should be able to steal focus from another of its own windows.

### CODE REVIEW

**Comment by wmww** [comment]
What the hell? That exact test (`a_second_window_of_different_app_leaves_first_active_when_unhiding`) passes on my machine.

**Review by RAOF — APPROVED** [review]
This seems like a reasonable behaviour, one the tests pass ;)

**Comment by wmww** [comment]
The tests that were broken seem to have been dropped from master, so we might be good to go.

CI seems to have ignored my push, hmm
bors try

File: src/miral/basic_window_manager.cpp [hunk 1]
```diff
@@ -1503,11 +1503,13 @@ void miral::BasicWindowManager::set_state(miral::WindowInfo& window_info, MirWin
         break;
 
     default:
-        auto const none_active = !active_window();
+        bool const can_become_active =
+            !active_window() ||
+            active_window().application() == window_info.window().application();
```
Comment (AlanGriffiths) [inline src/miral/basic_window_manager.cpp:1508]: Nit: `window_info.window()` is already cached as `window`

**Review by AlanGriffiths — APPROVED** [review]
bors delegate=wmww

File: src/miral/basic_window_manager.cpp (hunk 1 above)
Comment (wmww) [reply src/miral/basic_window_manager.cpp:1508]: So it is

**Pull request merged**